components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.zookeeper/src/main/java/org/wso2/carbon/cluster/coordinator/zookeeper/ZookeeperCoordinationStrategy.java

```diff
@@ -151,15 +162,19 @@ public ZookeeperCoordinationStrategy() throws IOException {
         client = CuratorFrameworkFactory
                 .newClient(this.zkURL, new ExponentialBackoffRetry(1000, 3));
         client.start();
-        cache = new PathChildrenCache(client, LEADER_ELECTION_ROOT_NODE + "/" + groupId, true);
+        cache = new PathChildrenCache(client, LEADER_ELECTION_ROOT_NODE + "/" + localGroupId, true);
         try {
             cache.start();
         } catch (Exception e) {
             throw new ClusterCoordinationException("Error while starting the path cache", e);
         }
-        String watchedNode = attemptForLeaderPosition(groupId, processNodePath);
-        addListener(cache, groupId, processNodePath, watchedNode,
-                getLeaderNode(groupId).getNodeId());
+        String watchedNode = attemptForLeaderPosition(localGroupId, processNodePath);
+        addListener(cache, localGroupId, processNodePath, watchedNode, getLeaderNode().getNodeId());
+    }
+
+    @Override
+    public void setPropertiesMap(Map<String, Object> propertiesMap) {
+        // TODO: 9/18/17
```

Comment:
Fix the TODO works or add required pending task details.

Reply:
Fixed in c5ceb96